We do not have the typed-ast package in the Userland yet. Since astroid can
work without it, we can safely remove it from its dependencies.

--- astroid-2.12.5/pyproject.toml
+++ astroid-2.12.5/pyproject.toml
@@ -35,7 +35,6 @@ dependencies = [
     "lazy_object_proxy>=1.4.0",
     "wrapt>=1.14,<2;python_version>='3.11'",
     "wrapt>=1.11,<2;python_version<'3.11'",
-    "typed-ast>=1.4.0,<2.0;implementation_name=='cpython' and python_version<'3.8'",
     "typing-extensions>=3.10;python_version<'3.10'",
 ]
 dynamic = ["version"]
